cocos/gfx/webgpu/instantiated.ts

```diff
@@ -0,0 +1,127 @@
+const registerList: any[] = [];
+
+function initWasm (wasmFactory, wasmUrl: string): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+        const errorMessage = (err: any): string => `[WebGPU]: WebGPU wasm load failed: ${err}`;
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        fetchUrl(wasmUrl).then((currUrl) => {
+            wasmFactory(currUrl).then((Instance: any) => {
+                wasmInstance = Instance;
+                registerList.forEach((cb) => {
+                    cb(wasmInstance);
+                });
+            }).then(resolve).catch((err: any) => reject(errorMessage(err)));
+        });
+    });
+}
+
+function initAsmJS (asmFactory, asmJsMemUrl: string): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+        fetchBuffer(asmJsMemUrl).then((arrayBuffer) => {
+            const wasmMemory: any = {};
+            wasmMemory.buffer = new ArrayBuffer(MEMORYSIZE);
+            const module = {
+                wasmMemory,
+                memoryInitializerRequest: {
+                    response: arrayBuffer,
+                    status: 200,
+                } as Partial<XMLHttpRequest>,
+            };
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+            return asmFactory(module).then((instance: any) => {
+                wasmInstance = instance;
+                registerList.forEach((cb) => {
+                    cb(wasmInstance);
+                });
+            });
+        }).then(resolve).catch(reject);
+    });
+}
+
+function shouldUseWasmModule (): boolean {
+    if (NATIVE_CODE_BUNDLE_MODE === (NativeCodeBundleMode.BOTH as number)) {
+        return sys.hasFeature(sys.Feature.WASM);
+    } else if (NATIVE_CODE_BUNDLE_MODE === (NativeCodeBundleMode.WASM as number)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+export async function waitForWebGPUWasmInstantiation (): Promise<void> {
+    const errorReport = (msg: any): void => {
+        error('Error during WebGPU Wasm instantiation:', msg);
+    };
+    try {
+        await ensureWasmModuleReady();
+        if (shouldUseWasmModule()) {
+            const [
+                glslModule,
+                glslWasmModule,
+                twgslModule,
+                twgslWasmModule,
+            ] = await Promise.all([
+                import('external:emscripten/webgpu/glslang.js'),
+                import('external:emscripten/webgpu/glslang.wasm'),
+                import('external:emscripten/webgpu/twgsl.js'),
+                import('external:emscripten/webgpu/twgsl.wasm'),
+            ]);
+            const glslFactory = glslModule.default;
+            const glslWasmUrl = glslWasmModule.default;
+            const twgslFactory = twgslModule.default;
+            const twgslWasmUrl = twgslWasmModule.default;
+            await initWasm(glslFactory, glslWasmUrl);
+            await initWasm(twgslFactory, twgslWasmUrl);
```

Comment:
We could load glsl and twgsl at the same time. They don't have any dependency, right ?
If so, we could use `Promise.all` to improve the loading.
```ts
await Promise.all([ initWasm(glslFactory, glslWasmUrl), initWasm(twgslFactory, twgslWasmUrl) ]);
```